components/style/properties/longhand/border.mako.rs

```diff
@@ -91,3 +91,749 @@ ${helpers.single_keyword("-moz-float-edge", "content-box margin-box",
+                SpecifiedValue::None => dest.write_str("none"),
+            }
+        }
+    }
+
+    #[inline]
+    pub fn get_initial_value() -> computed_value::T {
+        computed_value::T::None
+    }
+
+    impl ToComputedValue for SpecifiedValue {
+        type ComputedValue = computed_value::T;
+
+        #[inline]
+        fn to_computed_value(&self, context: &Context) -> computed_value::T {
+            match *self {
+                SpecifiedValue::Image(ref image) =>
+                    computed_value::T::Image(image.to_computed_value(context)),
+                SpecifiedValue::None => computed_value::T::None,
+            }
+        }
+        #[inline]
+        fn from_computed_value(computed: &computed_value::T) -> Self {
+            match *computed {
+                computed_value::T::Image(ref image) =>
+                    SpecifiedValue::Image(ToComputedValue::from_computed_value(image)),
+                computed_value::T::None => SpecifiedValue::None,
+            }
+        }
+    }
+
+    pub fn parse(context: &ParserContext, input: &mut Parser) -> Result<SpecifiedValue, ()> {
+        if input.try(|input| input.expect_ident_matching("none")).is_ok() {
+            return Ok(SpecifiedValue::None);
+        }
+
+        Ok(SpecifiedValue::Image(try!(Image::parse(context, input))))
+    }
+</%helpers:longhand>
+
+<%helpers:longhand name="border-image-outset" products="none" animatable="False">
+    use cssparser::ToCss;
+    use std::fmt;
+    use values::LocalToCss;
+    use values::NoViewportPercentage;
+    use values::specified::{Length, Number};
+
+    impl NoViewportPercentage for SpecifiedValue {}
+
+    pub mod computed_value {
+        use values::computed::{Length, Number};
+        #[derive(Debug, Clone, PartialEq)]
+        #[cfg_attr(feature = "servo", derive(HeapSizeOf))]
+        pub struct T(pub LengthOrNumber, pub LengthOrNumber,
+                     pub LengthOrNumber, pub LengthOrNumber);
+
+        #[derive(Debug, Clone, PartialEq)]
+        #[cfg_attr(feature = "servo", derive(HeapSizeOf))]
+        pub enum LengthOrNumber {
+            Length(Length),
+            Number(Number),
+        }
+    }
+
+    #[derive(Debug, Clone, PartialEq)]
+    #[cfg_attr(feature = "servo", derive(HeapSizeOf))]
+    pub struct SpecifiedValue(pub Vec<LengthOrNumber>);
+
+    impl ToCss for computed_value::T {
+        fn to_css<W>(&self, dest: &mut W) -> fmt::Result where W: fmt::Write {
+            try!(self.0.to_css(dest));
+            try!(dest.write_str(" "));
+            try!(self.1.to_css(dest));
+            try!(dest.write_str(" "));
+            try!(self.2.to_css(dest));
+            try!(dest.write_str(" "));
+            self.3.to_css(dest)
+        }
+    }
+    impl ToCss for SpecifiedValue {
+        fn to_css<W>(&self, dest: &mut W) -> fmt::Result where W: fmt::Write {
+            let length = self.0.len();
+
+            try!(self.0[0].to_css(dest));
+            if length > 1 {
```

Comment:
Just loop through `self.0.iter().skip(1)` emitting a space and the element each iteration.